pkg/metrics/metrics_controller.go

```diff
@@ -147,12 +146,10 @@ var (
 type Controller struct {
 	MetricsPath string
 	MetricsPort uint16
-	mu          sync.Mutex
-	nodeIP      net.IP
 }
 
 // Run prometheus metrics controller
-func (mc *Controller) Run(healthChan chan<- *healthcheck.ControllerHeartbeat, stopCh <-chan struct{}, wg *sync.WaitGroup) error {
+func (mc *Controller) Run(healthChan chan<- *healthcheck.ControllerHeartbeat, stopCh <-chan struct{}, wg *sync.WaitGroup) {
```

Comment:
Removed error return value here since it was not used and it's a go routine, so probably an error channel makes more sense.